Added Functions

shaz13 opened the pull request:
1. Anonymize - removes user names/ IP addresses from the comment
2. Apostophes - replaces are'nt with arenot etc

kamil-kaczmarek commented on steps/preprocessing.py:
```diff
@@ -3,12 +3,20 @@
 from sklearn.feature_extraction import text
 import sklearn.preprocessing as sk_prep
-
+from nltk.tokenize import word_tokenize
```
@shaz13 ,
This introduces additional dependency, that is `nltk` -> this should be added to the requirements.txt

shaz13 replied:
Done.

kamil-kaczmarek replied:
@shaz13 Nice thanks

shaz13 replied:
Done

kamil-kaczmarek commented on steps/preprocessing.py:
```diff
@@ -3,12 +3,20 @@
+lem = WordNetLemmatizer()
+tokenizer=TweetTokenizer()
+eng_stopwords = set(stopwords.words("english"))
+APPO = {"aren't": 'are not', "can't": 'cannot', "couldn't": 'could not', "didn't": 'did not', "doesn't": 'does not', "don't": 'do not', "hadn't": 'had not', "hasn't": 'has not', "haven't": 'have not', "he'd": 'he would', "he'll": 'he will', "he's": 'he is', "i'd": 'I had', "i'll": 'I will', "i'm": 'I am', "isn't": 'is not', "it's": 'it is', "it'll": 'it will', "i've": 'I have', "let's": 'let us', ...
```
APPO is quite large dict, thus - for clarity - it should be moved to the utils.py, with more descriptive name.
@shaz13 @jakubczakon what do you guys think?

shaz13 replied:
Right now. This is a raw approach. I think this could be handled by a clever regex statement. 
Something that replaces 
``n't --> not,  'd --> would ,  'll --> will , 'm --> am and 's with is`` should cover all words in dict. Let me know what you guys think @kamil-kaczmarek  @jakubczakon

kamil-kaczmarek replied:
@shaz13 I agree, that in principle clever regex would be final goal. However, for the first iteration your dict is good (it is not too large :) ). If you agree, with my idea of small refactor, you can move APPO to `utils.py` in the repo root, (not `steps/utils.py`). If not, let @jakubczakon make final decision on this.

kamil-kaczmarek commented on steps/preprocessing.py:
```diff
@@ -42,13 +50,15 @@ def save(self, filepath):
 class TextCleaner(BaseTransformer):
     def __init__(self, drop_punctuation, drop_newline, drop_multispaces,
-                 all_lower_case, fill_na_with, deduplication_threshold):
+                 all_lower_case, fill_na_with, deduplication_threshold, anonymize, apostophes):
```
Both `anonymize`, and `apostophes` are boolean, so it might be beneficial for the broader community to move it as a parameter to the `config.yaml`, to easily control this behavior.

If not it should be handled in the: [pipeline_config.py:L19](https://github.com/neptune-ml/kaggle-toxic-starter/blob/master/pipeline_config.py#L19)

shaz13 replied:
Implemented in `pipeline_config.py:L19` as of now

kamil-kaczmarek replied:
@shaz13 thanks, I can see, that you added two more params to `neptune.yaml` -> :+1:

shaz13 replied:
:+1:

shaz13 commented:
Note: The nltk in the code uses wordnet and stopwords list which should be priorly downloaded in the environment using `nltk.download('wordnet')` and `nltk.download('stopwords')`. Are these persistent in Neptune environment or each run requires the download again?

shaz13 commented:
@jakubczakon @kamil-kaczmarek 
Added functionality to choose using stopwords or not. Added nltk.downloads respectively. Awaiting decision on APPO dict placement

kamil-kaczmarek commented:
Hey @shaz13,
I have discussed APPO with @jakubczakon, and we have decided to do it in a clean way. That is:
1. APPO dict goes to `external_data/apostrophes.json`. Note that it is _json_ file,
2. In the `steps/preprocessing.py` you can load it to dict, using json module.

When this is done, we are ready to merge.

Sorry for iterating this multiple times - we just want to maintain clean implementation ;-)

Thanks!

kamil-kaczmarek commented:
@shaz13 @jakubczakon merge done.
